src/viadot/sources/mediatool.py

```diff
@@ -0,0 +1,334 @@
+"""'mediatool.py'."""
+
+import json
+
+import pandas as pd
+from pydantic import BaseModel
+
+from viadot.config import get_source_credentials
+from viadot.exceptions import APIError, CredentialError
+from viadot.sources.base import Source
+from viadot.utils import add_viadot_metadata_columns, handle_api_response
+
+
+class MediatoolCredentials(BaseModel):
+    """Checking for values in Mediatool credentials dictionary.
+
+    Two key values are held in the Mediatool connector:
+        - user_id: The unique ID for the organization.
+        token: A unique token to be used as the password for API requests.
+
+    Args:
+        BaseModel (pydantic.main.ModelMetaclass): A base class for creating
+            Pydantic models.
+    """
+
+    user_id: str
+    token: str
+
+
+class Mediatool(Source):
+    """Class implementing the Mediatool API.
+
+    Download data from Mediatool platform. Using Mediatool class user is able to
+    download organizations, media entries, campaigns, vehicles, and media types data.
+    """
+
+    def __init__(
+        self,
+        *args,
+        credentials: MediatoolCredentials | None = None,
+        config_key: str | None = None,
+        user_id: str | None = None,
+        **kwargs,
+    ):
+        """Create an instance of the Mediatool class.
+
+        Args:
+            credentials (MediatoolCredentials, optional): Meditaool credentials.
+                Defaults to None.
+            config_key (str, optional): The key in the viadot config holding relevant
+                credentials. Defaults to None.
+            user_id (str, optional): User ID. Defaults to None.
+        """
+        credentials = credentials or get_source_credentials(config_key) or None
+        if credentials is None:
+            message = "Missing credentials."
+            raise CredentialError(message)
+
+        validated_creds = dict(MediatoolCredentials(**credentials))
+        super().__init__(*args, credentials=validated_creds, **kwargs)
+
+        self.header = {"Authorization": f"Bearer {credentials.get('token')}"}
+        self.user_id = user_id or credentials.get("user_id")
+
+        self.url_abbreviation = None
+
+    def _rename_columns(
+        self,
+        df: pd.DataFrame,
+        column_suffix: str,
+    ) -> pd.DataFrame:
+        """Rename columns.
```

Comment:
Ok but you do it in some specific way, otherwise you would just use `df.rename()`. What's the logic here? Can you add an example here in docstring?

Reply:
I can't because it is not my code. It is just a transcription from the original. Create a backlog task with your concern.

Reply:
You're adding the code, so you are the owner. Ok, feel free to create an issue - I'm fine with you adding it in a separate PR.

Reply:
#1062